src/agent/src/rpc.rs

```diff
@@ -215,6 +218,14 @@ impl AgentService {
             "receive createcontainer, storages: {:?}", &req.storages
         );
 
+        // In case of pulling image inside guest, we need to merge the image bundle OCI spec
+        // into the container creation request OCI spec.
+        #[cfg(feature = "guest-pull")]
```

Comment:
It's important for us to log the capabilities of the agent so please can you update `announce()` to add an `agent-features` list which will potentially contain `guest-pull` and `agent-policy`, but be empty by default. Once that's done, the runtime will automatically log the agents features.

Reply:
Hi @jodh-intel ! I'm trying to understand your comment.

Did you mean to update https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/main.rs#L123 ? Or having an `announce()` for the rpc.rs file?

In either case, will it call `announce()` every time for `do_create_container()`?

Can we send that on a follow on PR?

Reply:
Hi @wainersm - What I mean is that the agent's `announce()` function (which is only called once, at agent startup) should display an `agent-features` list, so it's clear from the logs what features it was built with.

Cloud Hypervisor does something similar when run with `--version` **and** `-v`:

```bash
$ cloud-hypervisor --version -v
cloud-hypervisor v35.0-82-gffb9a051
Enabled features: ["io_uring", "kvm"]
```

See:

- https://github.com/cloud-hypervisor/cloud-hypervisor/blob/18a4d73/src/main.rs#L791
- https://github.com/cloud-hypervisor/cloud-hypervisor/blob/18a4d73/vmm/src/lib.rs#L299

But also, the `agent-features` should be added to the agent protocol file so that the `GetGuestDetails()` ttRPC API also returns that additional information:

- https://github.com/kata-containers/kata-containers/blob/442df71/src/libs/protocols/protos/agent.proto#L404

Reply:
On reflection, I don't think this PR needs to add that feature, so I've raised #9286. Once that lands, you can just update `src/agent/src/features.rs` for the `"guest-pull"` feature (and any others this PR adds).

Reply:
> On reflection, I don't think this PR needs to add that feature, so I've raised #9286. Once that lands, you can just update `src/agent/src/features.rs` for the `"guest-pull"` feature (and any others this PR adds).

Hi @jodh-intel thanks for the contribution. I plan to add the feature to annonce() once the PR has been merged: https://github.com/kata-containers/kata-containers/issues/9225